h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/rabit/RabitTrackerH2O.java

```diff
@@ -71,9 +71,8 @@ public boolean start(long timeout) {
             throw new IllegalStateException("Rabit tracker already started.");
         }
         RabitTrackerH2OThread trackerThread = new RabitTrackerH2OThread(this);
-        trackerThread.setDaemon(true);
-        trackerThread.start();
         this.trackerThread = trackerThread;
+        trackerThread.start();
```

Comment:
First assign it, then start it. If start() throws an unechecked exception, the pointer is still there in `this.trackerThread` and can be dealth with later.